great_expectations/execution_engine/sparkdf_execution_engine.py

```diff
@@ -74,6 +87,9 @@
         "Unable to load pyspark; install optional spark dependency for support."
     )
 
+if TYPE_CHECKING:
+    from pyspark.sql import DataFrame
```

Comment:
Why do we need to import this again?

Reply:
It's because otherwise DataFrame could be None because of the try imports... But this brings up another issue which is now we have to install pyspark to do type-checking which might not be desirable.

I think I'll revert this.